### DISCUSSION TITLE
如何将一张图片上传到q群上

### DISCUSSION THREAD

**Discussion by wc171177** [simple-robot/Discussions#92] [Q&A] [answered]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

如何将一张图片上传到q群上，看了好多接口都没有看到如何实现

**Comment by ForteScarlet** [comment, marked as answer]
- https://simbot.forte.love/docs/definition/message-overview/standard-message#image
- https://simbot.forte.love/docs/basic/messages#%E6%B6%88%E6%81%AF%E5%8F%91%E9%80%81
- https://docs.simbot.forte.love/components/mirai/simbot-component-mirai-core/love.forte.simbot.component.mirai.message/-mirai-send-only-image/index.html

**Reply by wc171177** [reply]
我少说了两个字，是如何把图片上传到q群的文件夹里面我少说了两个字，是如何把图片上传到q群的文件夹里面

**Reply by wc171177** [reply]
就是上传到群上的图片创建一个文件夹把图片放里面

**Reply by ForteScarlet** [reply]
https://github.com/orgs/simple-robot/discussions/31

**Reply by wc171177** [reply]
谢谢啦

**Reply by wc171177** [reply]
uploadNewFile  可以解释一下这个函数么    需要填写的不是很理解

**Reply by ForteScarlet** [reply]
这是mirai的API，自行阅读其注释或前往 [它们的API文档](https://kdoc.mirai.mamoe.net/) 搜索阅读

**Reply by wc171177** [reply]
好的

**Reply by wc171177** [reply]
上传的图片在群文件的根目录怎么办    用下面的方法也不可以
filepath：
或是绝对路径 /sub/foo.txt 表示根目录的 "sub" 目录下的文件 "foo.txt"

**Reply by ForteScarlet** [reply]
阅读文档。
https://kdoc.mirai.mamoe.net/2.14.0/mirai-core-api/net.mamoe.mirai.contact.file/-remote-files/index.html

**Reply by wc171177** [reply]
我弄不好   可以提示一下么
我用绝对路径 /sub/foo.txt的时候 Failed to create folder '/75': -303 文件夹名称错误.

**Reply by ForteScarlet** [reply]
`RemoteFiles` 是 mirai 的API，如果有疑问，前往它们的 [讨论区](https://github.com/mamoe/mirai/discussions) 、[issues](https://github.com/mamoe/mirai/issues/new/choose) 或 [论坛](https://mirai.mamoe.net/) ，我对此API并不熟悉。
